drivers/ethernet/eth_stm32_hal.c

```diff
@@ -1519,6 +1524,7 @@ static void eth_iface_init(struct net_if *iface)
 		/* Now that the iface is setup, we are safe to enable IRQs. */
 		__ASSERT_NO_MSG(cfg->config_func != NULL);
 		cfg->config_func();
+		k_sem_give(&dev_data->init_sem);
```

Comment:
Maybe, instead of introducing a new semaphore, just spawn the thread here instead of in `eth_initialize()`?

Reply:
Good idea, much simpler and works just as well.